packages/react-color-picker/src/lib/webgl.ts

```diff
@@ -96,17 +93,21 @@ in vec4 v_end_color;
 out vec4 f_color;
 
 ${LINEAR_RGB}
+${OKLAB}
 
 void main() {
   float a = vec2(gl_FragCoord.xy / v_resolution).x;
-  f_color = blend_srgb(v_start_color, v_end_color, a);
+  f_color = linear_rgb_to_srgb(avg_vec4(oklab_to_linear_rgb(v_start_color), oklab_to_linear_rgb(v_end_color), a));
 }
 `;
 
-export class GradientRGB {
+/**
+ * Create a gradient from A to B, blended in Oklab space.
+ */
+export class GradientOklab {
   gl: WebGL2RenderingContext;
-  startColor: WebGLColor;
-  endColor: WebGLColor;
+  startColor: Oklab;
```

Comment:
Another design problem of the WebGL code was trying to make it magically handle all color spaces, all gamuts. This is far too much responsibility for it, and also isn’t necessary since at any given time, we’re only operating in one colorspace!

Instead, if we simply restrict the WebGL code to operate in the Oklab space alone, we don’t lose any performance benefits. It does what it does best—performs crazy-fast translations between Oklab <> Linear RGB. And the only time we have to calculate colorspaces on the main thread in JS is just 2 times—start and end color, which is trivial. Keeping WebGL in Oklab still reaps all the benefits but sheds a ton of complexity and potential bugs.